Some additions to generate.py after #1948

h-mayorquin opened the pull request:
Some additiosn after #1948.

* In `generate_recording`. We say that we will deprecate legacy in the next release but we don't have a deprecation warning. Next release is 0.99, so when this is released it should start throwing a deprecation warning, then we can deprecate it in 0.100. I added the deprecation warning. Plus, if we are going to deprecate we should make lazy the default in main so the code that we start suggestin people to use from main remains valid. Otherwise, they we will need to keep an argument with only one possible value (mode="lazy").
* I added the type directly in `rng.standard_normal` as this was generated an expensive computation when I profiled (got 30 % increase in performance for a one hour recorder with "on_the_fly")
* I also modified the default value for the standard deviation in the NoiseGenerator to 1. I think is a better [Schelling Point.](https://en.wikipedia.org/wiki/Focal_point_(game_theory)).
* I also modified the number of channels in `generate_recording_by_size` to be the same as SpikeGLX. I remember that at some point it caused me some problems in testing for it to have such an incovenient value (1024) which I think I set because it will play nice with round number of seconds at the beginning but it does not make a lot of sense in retrospective.

h-mayorquin commented on src/spikeinterface/core/generate.py:
```diff
@@ -586,35 +591,35 @@ def get_traces(
 
         pos = 0
-        for block_index in range(start_block_index, end_block_index + 1):
+        for block_index in range(first_block_index, last_block_index + 1):
```
It was very strange that this version is faster than my vectorized one. Really surprised about this. That said, I realized that I had a some bad variable names which I have corrected here.

samuelgarcia replied:
I think we can be even faster in we would use the "out" the rng.standard_normal but the code logic between the 2 modes could not be shared. So lets keep this.
For testing the default mode should be always "tile_pregenerated"

h-mayorquin replied:
When I profiled it was showing the `np.concatenate` was the culprit of it being so slow. I will check out on what you say, I remember that I thought about it.

samuelgarcia commented on src/spikeinterface/core/generate.py:
```diff
@@ -503,7 +508,7 @@ def __init__(
         sampling_frequency: float,
         durations: List[float],
-        noise_level: float = 5.0,
+        noise_level: float = 1.0,
```
I think 5 uV is more realistic by default no ?

alejoe91 replied:
I agree with @samuelgarcia !

h-mayorquin replied:
Color me skeptical aoubt a consensus in the size of noise for electrophysiology across areas, devices, species? (totally curios to be proven wrong). Do we want to make a claim like that?

That said, even if there was a consensus I think it would make more sense to put it in the downstream functions that are  mean to generate realistic recordings like `generate_ground_truth_recording` and keep this as a general generator that can be used to create different noise scenarios. For that, I think it makes more sense to keep this as the standard white noise instead of trying to make it realistically. For example, if useres wanted to generate pink noise I think it will be easier to do it from the base white noise instead of a scaled version of it.

samuelgarcia replied:
lets keep this the generate_ground_truth_recording have already 5

samuelgarcia commented:
Merci beaucoup.

> Some additiosn after #1948.
> 
> * In `generate_recording`. We say that we will deprecate legacy in the next release but we don't have a deprecation warning. Next release is 0.99, so when this is released it should start throwing a deprecation warning, then we can deprecate it in 0.100. I added the deprecation warning. Plus, if we are going to deprecate we should make lazy the default in main so the code that we start suggestin people to use from main remains valid. Otherwise, they we will need to keep an argument with only one possible value (mode="lazy").

My plan was to do this in 2 steps first keep "legacy" by default and the switch to lazy. I am ok with lazy now.

> * I added the type directly in `rng.standard_normal` as this was generated an expensive computation when I profiled (got 30 % increase in performance for a one hour recorder with "on_the_fly")

Of course this avoid one amalooc! Really cool. thank you!

> * I also modified the default value for the standard deviation in the NoiseGenerator to 1. I think is a better [Schelling Point.](https://en.wikipedia.org/wiki/Focal_point_(game_theory)).

If we do this. I would like to still have 5 in generate_ground_truth_recording but I think this already the case.

> * I also modified the number of channels in `generate_recording_by_size` to be the same as SpikeGLX. I remember that at some point it caused me some problems in testing for it to have such an incovenient value (1024) which I think I set because it will play nice with round number of seconds at the beginning but it does not make a lot of sense in retrospective.

Not sure we have to mimic the neuropixel always. sometimes it is good to change number for testing.
But in fact, I do not care so much.

h-mayorquin commented:
> > * I also modified the default value for the standard deviation in the NoiseGenerator to 1. I think is a better [Schelling Point.](https://en.wikipedia.org/wiki/Focal_point_(game_theory)).
> 
> If we do this. I would like to still have 5 in generate_ground_truth_recording but I think this already the case.

Great. I think this makes sense.

> > * I also modified the number of channels in `generate_recording_by_size` to be the same as SpikeGLX. I remember that at some point it caused me some problems in testing for it to have such an incovenient value (1024) which I think I set because it will play nice with round number of seconds at the beginning but it does not make a lot of sense in retrospective.
> 
> Not sure we have to mimic the neuropixel always. sometimes it is good to change number for testing. But in fact, I do not care so much.

Yeah, I agree with you, it is just a better default than the one I had before which I chosen because of numerology (e.g. 1024 looks nice)!
